server/system/user-authenticator.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <functional>
+#include <list>
+#include <string>
+
+#include "data-source.hpp"
+#include "user/user.hpp"
+
+namespace tec {
+using namespace user;
+namespace system {
+class UserAuthenticator {
```

Comment:
Used as a way to abstract away user authentication. Right now the only data source is the save game's user list, but this could presumably be a database of some kind.